k8s/cloud_deps/base/nats/statefulset.yaml

```diff
@@ -225,3 +228,44 @@ spec:
               # the NATS Server to gracefully terminate the client connections.
               #
               command: ["/bin/sh", "-c", "/nats-server -sl=ldm=/var/run/nats/nats.pid && /bin/sleep 60"]
+
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+          runAsNonRoot: true
+          runAsUser: 10100
+          seccompProfile:
+            type: RuntimeDefault
+
+      initContainers:
+      - name: init-chown-data
```

Comment:
Would it be possible to use `fsGroup` and `fsGroupChangePolicy` to accomplish this rather than an initContainer?

Reply:
I tested with `fsGroup` for with both nats and postgres workloads. Using `fsGroup` for nats server worked as expected.

But adding the it thing for postgres deployment result in an error stated as follows.
```
chmod: changing permissions of '/var/lib/postgresql/data': Operation not permitted
chmod: changing permissions of '/var/run/postgresql': Operation not permitted

FATAL:  data directory "/var/lib/postgresql/data" has wrong ownership 
```

postgres user within the container has the id of `999` yet having the following security context did fix the issue.

```
securityContext:
  fsGroup: 999
  runAsGroup: 999
  runAsNonRoot: true
  fsGroupChangePolicy: "Always"
  runAsUser: 999
  seccompProfile:
    type: RuntimeDefault
```

Reply:
I believe the `postgres` user is 70.
```
docker run --rm -it postgres:14-alpine id -u postgres
70
```

And it seems to work with the uid set to 70. So let's use that instead?

Reply:
The user is `999` under bullseye but we use alpine postgres which uses `70` :)